user-interface/0005-ETXN-submit_ethereum_transaction.md

```diff
@@ -1 +1,56 @@
-# Submit ethereum transaction
\ No newline at end of file
+# Submit Ethereum transaction
+
+## Know what transaction I'm signing
+When about to click to prompt Ethereum wallet to sign a transaction, I...
+
+- **should** see the contract address I am about to interact with
+- **should** see the function name I am about to interact with
+
+...so I know what to expect when my wallet asks me to sign
+## Track transactions to wallet
+
+after clicking to submit an eth transaction to a connected wallet, I...
+
+- **could** see an estimate for gas prices compared to a recent history
+- **could** see an estimated gas price for the function in question
+- **must** see prompt to check Ethereum wallet to approve transactions
+
+... so I know I need to go to my wallet app to approve the transaction
+
+## ERC20 approval/permit
+
+> The approval/permit step is part of the [ERC20 standard](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/). It's intention is to provide additional security when interacting with smart contracts / using Dapps. It sets a maximum amount that the given eth key can send to a given address. It means that before end "spend" of ERC20 tokens I will have need to submit a "approve" transaction to tell the network how much is approved. An attempt to spend more than the approved amount will fail.
+> For example: In my approve transaction, I approve 10, I then spend 5. In my next transaction I attempt to spend 6, this will fail as my approve amount was reduced by the first transaction.
+> It is common for dapps to have a approve button that simply asks to approve a massive amount and leave it up to the wallet UI (e.g. metamask) to ask the user if they would like to change this. 
+> Some ERC20 contracts have an additional function that runs both the approve and deposit function in one, but this is not standard.
+
+If the transaction in question requires an ERC20 approval, I...
+
+- if the current approved amount is less than the amount being "spent": **must** see be prompt to approve 
+- **could** see the current approved amount
+- **should** be able to set the amount to be approved (incase the connected wallet does not handle this)
+- **must** send an approve transaction with either a user specified amount or a very large number (<a name="0005-ETXN-001" href="#0005-ETXN-001">0005-ETXN-001</a>)
```

Comment:
'very large number', seems a bit dubious. I think the must should actually just be user approved only, even though we don't do this currently

Reply:
Agree about making a user set their own amount a must. https://github.com/vegaprotocol/specs/pull/1300/commits/39d3bbb3c5eaa1f04f839dae4ba5e6cba61a27c2 

But I don't agree we should do something different to the majority of De-fi. I don't think I've ever seen a dapp that doesn't just approve a massive amount and leave it to the user to change in their wallet. (but I do agree it is dubious).

Perhaps it could just be a big number and we tell the user this is the default.
@mattrussell36  What is the huge number at the moment? I'll add it to ACs

Reply:
Hm okay, I think this is fine then. Lets discuss what that value should be. Right now it can vary slightly